### PR TITLE
[BEAM-151] Move Dataflow runner to new Dataflow runner maven module

### PR DESCRIPTION
**Pull request by lukecwik**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace "<Jira issue #>" in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
Note that Flink runner and Java examples both depend on the Dataflow runner still. So I needed to make the Dataflow runner a top level component to satisfy these dependencies.

Also, we are currently shading within the Dataflow runner and using the same namespace to reduce the jar size.

### CODE REVIEW

**Comment by lukecwik** [comment]
R: @davorbonaci 
CC: @jbonofre 
CC: @amitsela

File: runners/google-cloud-dataflow-java/pom.xml [hunk 1]
```diff
@@ -0,0 +1,557 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+      <groupId>org.apache.beam</groupId>
+      <artifactId>parent</artifactId>
+      <version>0.1.0-incubating-SNAPSHOT</version>
+      <relativePath>../../pom.xml</relativePath>
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/pom.xml]: Move to its regular parent. (If not possible, add comments around it.)

File: runners/google-cloud-dataflow-java/pom.xml [hunk 2]
```diff
@@ -0,0 +1,557 @@
...
+  </parent>
+
+  <artifactId>google-cloud-dataflow-java-runner</artifactId>
+
+  <name>Apache Beam :: Runners :: Google Cloud Dataflow for Java</name>
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/pom.xml]: rm "for Java"

File: runners/google-cloud-dataflow-java/pom.xml [hunk 3]
```diff
@@ -0,0 +1,557 @@
...
+      <groupId>org.apache.beam</groupId>
+      <artifactId>java-sdk-all</artifactId>
+      <version>${beam.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>org.apache.beam</groupId>
+      <artifactId>java-sdk-all</artifactId>
+      <version>${beam.version}</version>
+      <type>test-jar</type>
```
Comment (davorbonaci) [inline runners/google-cloud-dataflow-java/pom.xml]: Move test scope to the bottom

**Comment by davorbonaci** [comment]
Nice! Just a few comments.

**Comment by davorbonaci** [comment]
(Also, I expect Coveralls to report a slight decrease in coverage; not sure if that is some flakiness, so somehow we report less.)

File: runners/google-cloud-dataflow-java/pom.xml (hunk 1 above)
Comment (lukecwik) [inline runners/google-cloud-dataflow-java/pom.xml]: Done

File: runners/google-cloud-dataflow-java/pom.xml (hunk 2 above)
Comment (lukecwik) [inline runners/google-cloud-dataflow-java/pom.xml]: Done

File: runners/google-cloud-dataflow-java/pom.xml (hunk 3 above)
Comment (lukecwik) [inline runners/google-cloud-dataflow-java/pom.xml]: Done

**Comment by davorbonaci** [comment]
LGTM (self-merge).

**Pull request closed without merging**